src/uv_secure/dependency_checker/dependency_checker.py

```diff
@@ -54,6 +54,23 @@
 
 USER_AGENT = f"uv-secure/{__version__} (contact: owenrlamont@gmail.com)"
 
+class DependencyFileParseException(Exception):
+    """Raised when a dependency file cannot be parsed correctly."""
+    pass
+
+
+class PackageMetadataProcessingException(Exception):
+    """Raised when processing package metadata fails."""
+    def __init__(self, package_name: str, original_exception: Exception):
+        super().__init__(f"{package_name} raised exception: {original_exception}")
+        self.package_name = package_name
+        self.original_exception = original_exception
```

Comment:
I don't know that you need to have a reference to the original exception like this... if you catch and rethrow one exception from another exception handler should should see both stack traces so I think it's unnecessary to explicitly link the exceptions like this.